echion/threads.h

```diff
@@ -294,6 +303,72 @@ void ThreadInfo::unwind_tasks()
     }
 }
 
+// ----------------------------------------------------------------------------
+void ThreadInfo::unwind_greenlets(PyThreadState* tstate)
```

Comment:
Why is it ok even though we don't acquire `greenlet_info_map_lock` in this function?

Reply:
Ah no I think we need to acquire the lock whenever we touch these maps, good callout!